modules/snooker/src/main/ts/ephox/snooker/calc/Deltas.ts

```diff
@@ -58,7 +63,20 @@ const determine = function (input: number[], column: number, step: number, table
     return onChange(index, next);
   };
 
+  // Applies to the last column bar
   const onRight = function (_prev: number, index: number) {
+    // 'default' means that the whole table and its columns need to be resized to maintain ratio
+    if (columnResizeBehaviour === 'default') {
```

Comment:
I don't really like all this branching code, as it makes it hard to follow/more complex. It'd be better if we can pass an algorithm in that takes the required values and returns the relevant result (similar to how TableSize works).

Reply:
Mm, I think we've all been talking about potentially doing that in various separate conversations? I'm definitely for it

Reply:
Would tie into the "init like Cement" idea?